cross-chain/polygon/deploy_sidechain/01_deploy_polygon_tbtc_token.ts

```diff
@@ -0,0 +1,43 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+import type { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { ethers, getNamedAccounts, helpers } = hre
+  const { deployer } = await getNamedAccounts()
+
+  const [, proxyDeployment] = await helpers.upgrades.deployProxy(
+    "PolygonTBTC",
+    {
+      contractName: "@keep-network/tbtc-v2/contracts/l2/L2TBTC.sol:L2TBTC",
+      initializerArgs: ["Polygon tBTC v2", "tBTC"],
+      factoryOpts: { signer: await ethers.getSigner(deployer) },
+      proxyOpts: {
+        kind: "transparent",
+      },
+    }
+  )
+
+  // TODO: Investigate the possibility of adding Tenderly verification for the
+  // sidechain and upgradable proxy.
+
+  // Contracts can be verified on Polygonscan in a similar way as we
+  // do it on L1 Etherscan
+  if (hre.network.tags.polygonscan) {
+    // Polygonscan might not include the recently added proxy transaction right
+    // after deployment. We need to wait some time so that transaction is
+    // visible on Polygonscan.
+    await new Promise((resolve) => setTimeout(resolve, 10000))
```

Comment:
Wouldn't it make more sense then to make the verification a separate step?

In case this timeout value is not enough, we'd be able to retry just the verification without the risk of deploying proxy again. In theory, the deploy plugin should detect the proxy is already deployed but better be safe than sorry.

Reply:
Yeah, we can do the verification as a separate step, but the timeout still be there. 10sec was always enough in Mumbai, but I don't know what it's going to be on mainnet.

Reply:
Why do we need a timeout if it is a separate step? Isn't it enough to wait with executing it until we see the transaction on Polygonscan? Keep in mind that for mainnet we execute steps manually one-by-one.

Reply:
Ok, but what about the testnet? Don't we want to execute deployment in a single command including verification? I can put if/else statement checking if this is a testnet and if it is, then wait 10sec.

Reply:
Here's the change [accb791](https://github.com/keep-network/tbtc-v2/pull/588/commits/accb79171df01922f2a95c6f73e660f4966d56da)